docarray/index/backends/in_memory.py

```diff
@@ -85,6 +87,8 @@ def __init__(
                     cast(Type[BaseDoc], self._schema)
                 )()
 
+        self._embedding_map: Dict[str, Tuple[AnyTensor, Optional[List[int]]]] = {}
```

Comment:
In the _extract_embeddings function, the return value is a tuple of embeddings and a list (optionally) of non-null indices to support empty value input. To avoid computing the non-null indices for each query, I have included a tuple containing the embeddings and the corresponding non-null indices.

Reply:
okey makes sense